src/search.rs

```diff
@@ -2,6 +2,27 @@
 
 use crate::is_prime;
 
+ 
+// Generalised function for nearest search by incrementing/decrementing by 1
+// Any attempt at optimising this would be largely pointless since the largest prime gap under 2^64 is only 1550
+// And is_prime's trial division already eliminates most of those
+const fn bounded_search(mut n: u64, stride: u64) -> Option<u64>{
```

Comment:
Would it be a good idea to use an enum for `stride` instead of a `u64` so that the function can be sure that the additions do what they are supposed to?

For example:

```rust
const fn bounded_search(mut n: u64, stride: Stride) -> Option<u64> {
    let stride = stride.into_u64();
    ...
}

#[derive(Debug, Clone, Copy)]
enum Stride {
    Up,
    Down,
}

impl Stride {
    const fn into_u64(self) -> u64 {
        match self
            Self::Up => 1,
            Self::Down => u64::MAX,
}
```

Reply:
Yeah, I was thinking that it would prevent the mistake of calling e.g. `bounded_search(n, 2)`.